lib/auth/cognito_stub_client.rb

```diff
@@ -64,9 +82,19 @@ def self.switch_to_stub
     return false if SelfService.service(:cognito_stub)
 
     real_client = SelfService.service(:cognito_client)
+    real_jwks = SelfService.service(:jwks)
     SelfService.register_service(name: :real_client, client: real_client)
+    SelfService.register_service(name: :real_jwks, client: real_jwks)
     SelfService.register_service(name: :cognito_client, client: stub_client)
+    load_jwks
     setup_stubs
     SelfService.register_service(name: :cognito_stub, client: true)
   end
+
+  def self.load_jwks
+    private_key = OpenSSL::PKey::RSA.generate(2048)
+    SelfService.register_service(name: :private_key, client: private_key)
+    jwks = JSON::JWK::Set.new(keys: [JSON::JWK.new(SelfService.service(:private_key).public_key, kid: 2)])
+    SelfService.register_service(name: :jwks, client: JSON.parse(jwks.to_json))
```

Comment:
is double JSON parsing neccessary?

Reply:
yes... one turns it in to a full JSON string and which you then have to parse in to a JSON Object.